File: real_time_chat/chat_socket_server/models/user_model.js
const { Schema, model } = require('mongoose');

const UserShema = Schema({
    name: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true,
        unique: true
    },
    password: {
        type: String,
        required: true
    },
    online: {
        type: Boolean,
        default: false
    }
});

UserShema.method('toJSON', function () {
    const { __v, _id, password, ...user } = this.toObject();
    user.uid = _id;
    return user;
});
module.exports = model('UserModel', UserShema)

File: real_time_chat/real_time_chat_app/lib/models/user_model.dart
class UserModel {
  bool? online;
  String? email;
  String? name;
  String? uid;

  UserModel({this.online, this.email, this.name, this.uid});

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
        online: json['online'] as bool?,
        email: json['email'] as String?,
        name: json['name'] as String?,
        uid: json['uid'] as String?);
  }

  Map<String, dynamic> toJson() {
      return {
        'online': online?? false,
        'email': email,
        'name': name,
        'uid': uid
      };
  }

  @override
  String toString() {
    return 'UserModel{online: $online, email: $email, name: $name, uid: $uid}';
  }
}
